Please use Requests, not urllib2

nwithan8 opened the pull request:
Urllib2 is old and deprecated.

Welcome to Python 🎉

nwithan8 committed:
Please, DO NOT use urllib2 in 2020.

nwithan8 committed:
Added requirements.txt

ReenigneArcher commented:
Thanks for the contribution! I was planning to replace urllib2 with requests as well. I initially used it as I had some functions that wouldn't work in python 3, those have since been removed though.

Let me review your changes.

nwithan8 commented:
Just warning you, I've spent the last two hours refactoring your Plex script, and I plan on doing the same for the others.
I don't mean to sound demeaning, I'm genuinely curious: How much experience do you have with Python? I just see some things here that I used to do in my earlier days starting out.

ReenigneArcher commented:
I'm not a programmer by any means, just a hobby.

I started using python this year. My experience is limited to these scripts, and a few metadata agent plugins I wrote for plex. Still learning.

nwithan8 commented:
Welcome!

ReenigneArcher commented on webm3u2m3u.py:
```diff
@@ -26,8 +26,8 @@ def quote_remover(string):
-    if input.startswith('http') == True: #https://www.tutorialspoint.com/python/string_startswith.htm
-        data = urllib2.urlopen(input) #https://stackoverflow.com/a/1393367/11214013
+    if input.startswith('http'): # https://www.tutorialspoint.com/python/string_startswith.htm
+        data = requests.get(url=input).text
```
Causes an issue at line 41. ```for line in data:```

No longer parses the input file properly. Returns a single character instead of entire line.

ReenigneArcher commented on nextpvr2m3u.py:
```diff
@@ -43,25 +43,18 @@ def GetListOfSubstrings(stringSubject,string1,string2):
-        result = json.loads(f.read())
-        return result
+
+    def load_url(url: str):
```
Please update this file based on latest in master (had to remove unnecessary imports and functions).

ReenigneArcher commented on plex.py:
```diff
@@ -26,13 +25,9 @@ def quote_remover(string):
+        return res.json()
     
     def fix(text):
         text = cgi.escape(text).encode('ascii', 'xmlcharrefreplace') #https://stackoverflow.com/a/1061702/11214013
```
```
Traceback (most recent call last):
  File ".\ReenigneArcher\web2tv-dev\plex.py", line 336, in <module>
    type = fix(grid['MediaContainer']['Metadata'][y]['type'])
  File ".\ReenigneArcher\web2tv-dev\plex.py", line 33, in fix
    text = cgi.escape(text).encode('ascii', 'xmlcharrefreplace') #https://stackoverflow.com/a/1061702/11214013
AttributeError: module 'cgi' has no attribute 'escape'
```

ReenigneArcher commented:
None of these updated files run properly. I need to review more in depth.

ReenigneArcher commented on plutotv2m3u.py:
```diff
@@ -1,7 +1,8 @@
 # -*- coding: utf-8 -*-
```
file replaced by plutotv.py

ReenigneArcher commented on plutotv2xml.py:
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
```
file replaced by plutotv.py

ReenigneArcher committed:
Update and rename webm3u2m3u.py to m3u_modder.py

-Replaced urllib2 with requests.
-Fixed issue with `for line in data:` going character by character, instead of line by line when using an m3u from a url.

ReenigneArcher committed:
Update and rename nextpvr2m3u.py to nextpvr.py

-Reverted functions to those provided by NextPVR developer.
-Replaced urllib2 with requests.
-Removed unneeded functions and imports.

ReenigneArcher committed:
Delete plutotv2m3u.py

This file was merged so epg and m3u generation for plutotv are contained in a single python script.

ReenigneArcher committed:
Update and rename plutotv2xml.py to plutotv.py

-Use requests instead of urllib2
-Added merged xml + m3u changes from master branch

Need to resolve an encoding issue when writing the xml file. M3u file appears to be okay.

ReenigneArcher commented on plutotv.py:
```diff
@@ -381,170 +409,217 @@ def fix(text):
+        #xml = xml.decode('unicode_escape')
+        #xml = xml.encode().decode('unicode_escape')
+        print('xml is ready to write')
+        #print(xml)
```
@nwithan8

If I uncomment line 603, the xml appears to print fine inside the command prompt window.

I cannot get it to write the file in line 608, unless I add the utf-8 encoding. Problem is it converts `&amp;` (and other symbols) back to `&`.

I want it to write the file exactly as I print it in line 603.

I am lost on this one... any idea?

nwithan8 replied:
Change "w" in the open(filename) function to "wb" to write bytes data rather than string data. Then don't convert the bytes to a string before writing to the file

ReenigneArcher committed:
Update requirements.txt

Added python-dateutil

ReenigneArcher committed:
Update plex.py

-Replaced urllib2 with requests
-Replaced cgi with html
-Removed json import

There is an encoding issue that needs to be resolved for the m3u. Xml appears to be okay.

ReenigneArcher committed:
Update plutotv.py

Fixed encoding in xml and m3u

ReenigneArcher committed:
Update plutotv.py

-Fixed quote escaping issue.
-Fixed math operation being returned as float instead of integer in Python 3.

Generated xml is now extremely close to the previous Python 2 version of this script.

ReenigneArcher committed:
Update plex.py

-Fixed quote escaping

ReenigneArcher committed:
Merge branch 'main' into main
